schema_based_on_diagram.sql

```diff
@@ -46,8 +52,29 @@ CREATE TABLE IF NOT EXISTS invoice_items
     quantity INT,
     total_price DECIMAL,
     invoice_id INT,
-    treatment_id INT
-)
+    treatment_id INT,
+    CONSTRAINT fk_invoces FOREIGN KEY(invoice_id) REFERENCES invoices(id) 
+            ON DELETE CASCADE
+            ON UPDATE CASCADE,
+    CONSTRAINT fk_treatments FOREIGN KEY(treatment_id) REFERENCES treatments(id)
+            ON DELETE CASCADE
+            ON UPDATE CASCADE
+);
+
+-- Join table between treatments and medical_histories
+DROP TABLE IF EXISTS treatment_medical_histories;
+CREATE TABLE IF NOT EXISTS treatment_medical_histories
+(
+    treatment_id INT,
+    medical_history_id INT,
+    CONSTRAINT fk_treatments FOREIGN KEY(treatment_id) REFERENCES treatments(id) 
+            ON DELETE CASCADE
+            ON UPDATE CASCADE,
+    CONSTRAINT fk_medical_histories FOREIGN KEY(medical_history_id) REFERENCES medical_histories(id) 
+            ON DELETE CASCADE
+            ON UPDATE CASCADE
+);
+
 
```

Comment:
- [ ] It would be better if all foreign keys have indexes created. For example; `CREATE INDEX ON medical_histories (patient_id);`. Indexes help us make a fast and quick query search

Reply:
This change has been done correctly